### PR TITLE
Add Github actions to run tests

### PR DESCRIPTION
**Pull request by Makonike**
## What My PullRequest Does
- Add e2e test in github actions.
- Replace fmt debug output with filter.callbacks.Log.
- Update README.md and some comments.
- Fix: Merge Config TODO

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -84,7 +84,14 @@ func (p *parser) Parse(any *anypb.Any) (interface{}, error) {
 }
 
 func (p *parser) Merge(parent interface{}, child interface{}) interface{} {
-	panic("TODO")
+	parentConfig := parent.(*config)
+	childConfig := child.(*config)
+
+	newConfig := *parentConfig
+	if childConfig != nil {
+		newConfig = *childConfig
```
Comment (doujiang24) [inline config.go:92]: better do merge logic, use the key values in child to override the newConfig.

File: Makefile [hunk 2]
```diff
@@ -1,30 +1,54 @@
...
+		--name envoy-ldap-test \
+		envoyproxy/envoy:contrib-v1.26.1 \
+		envoy -c /etc/envoy/envoy.yaml &
+	sleep 5
+	go test test/e2e_search_test.go
+
+	docker rm -f envoy-ldap-test
+
+test-bind:
+	go test test/e2e_search_test.go
```
Comment (doujiang24) [inline Makefile:48]:
```suggestion
	go test test/e2e_bind_test.go
```
the ci is still passed, I'm a bit confused.

Comment (Makonike) [reply Makefile]: It is provided for user testing purposes. I made a mistake here, and it has been deleted now.

**Review by doujiang24 — APPROVED** [review]

File: config.go (hunk 1 above)
Comment (doujiang24) [reply config.go:92]: oh, I missed this one.

**Review by doujiang24 — CHANGES_REQUESTED** [review]

File: config.go (hunk 1 above)
Comment (Makonike) [reply config.go]: Done.

File: Makefile (hunk 2 above)
Comment (Makonike) [reply Makefile]: Done.

**Review by doujiang24 — APPROVED** [review]

**Pull request merged**